### PR TITLE
Bg submit feedback

### PR DESCRIPTION
**Pull request by BhavyaGulati**
Closes #X

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

**Comment by BhavyaGulati** [comment]
[(Bhavya)React Issue 3: Submit Feedback to backend](https://trello.com/c/7XnQZF80/78-bhavyareact-issue-3-submit-feedback-to-backend)

File: app/javascript/image_sharing/stores/FeedbackStore.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { action, observable} from "mobx";
 export default class FeedbackStore {
   @observable name = "";
   @observable feedback = "";
+  @observable alertMessage = "";
```
Comment (butzhang) [inline app/javascript/image_sharing/stores/FeedbackStore.js]: do you have test to cover this line?

File: app/javascript/image_sharing/stores/FeedbackStore.js [hunk 2]
```diff
@@ -12,5 +13,9 @@ export default class FeedbackStore {
   @action setFeedback = (feedback) => {
     this.feedback = feedback;
   };
+
+  @action setAlertMessage = (alertMessage) => {
```
Comment (butzhang) [inline app/javascript/image_sharing/stores/FeedbackStore.js]: same here. do you have test to cover this line?

File: app/javascript/image_sharing/services/PostFeedbackService.js [hunk 3]
```diff
@@ -1,7 +1,25 @@
...
+  constructor(feedbackStore) {
+    this.feedbackStore = feedbackStore;
+  }
+
+  postFeedback = () => {
+    return post('/api/feedbacks', {feedback: {
+        name: this.feedbackStore.name,
+        feedback: this.feedbackStore.feedback
+      }}).then(
+      () => {0
```
Comment (butzhang) [inline app/javascript/image_sharing/services/PostFeedbackService.js]: we have a `0` here, maybe you want to remove it

File: app/javascript/image_sharing/components/FeedbackForm.js [hunk 4]
```diff
@@ -33,6 +33,7 @@ class FeedbackForm extends Component {
   render() {
     return (
       <div className="formGroup">
+        <p className="alert-message">{this.props.stores.feedbackStore.alertMessage}</p>
```
Comment (butzhang) [inline app/javascript/image_sharing/components/FeedbackForm.js]: can you try using `alert` from `react-gears`?

File: app/javascript/image_sharing/stores/FeedbackStore.js (hunk 2 above)
Comment (BhavyaGulati) [reply app/javascript/image_sharing/stores/FeedbackStore.js]: nope, fixing!

File: app/javascript/image_sharing/components/FeedbackForm.js [hunk 5]
```diff
@@ -33,6 +33,7 @@ class FeedbackForm extends Component {
   render() {
     return (
       <div className="formGroup">
+        {this.props.stores.feedbackStore.alertMessage? <Alert color="info" className="alert-message">{this.props.stores.feedbackStore.alertMessage}</Alert> : null}
```
Comment (butzhang) [inline app/javascript/image_sharing/components/FeedbackForm.js]: you can also try to make `this.props.stores.feedbackStore.alertMessage` as a function in the store with `computed` decorator.

Comment (BhavyaGulati) [reply app/javascript/image_sharing/components/FeedbackForm.js]: Cool! thanks

File: app/javascript/image_sharing/test/stores/FeedbackStore.spec.js [hunk 6]
```diff
@@ -21,4 +22,9 @@ describe('FeedbackStore', () => {
     feedbackStore.setFeedback('Go team!');
     expect(feedbackStore.feedback).to.equal('Go team!');
   });
+
+  it('setAlertMessage', () => {
+    feedbackStore.setAlertMessage('Feedback submitted');
+    expect(feedbackStore.alertMessage).to.equal('Feedback submitted');
```
Comment (butzhang) [inline app/javascript/image_sharing/test/stores/FeedbackStore.spec.js]: you need to do `expect(feedbackStore. AlertMessage).to.equal('Feedback submitted');`

Comment (BhavyaGulati) [reply app/javascript/image_sharing/test/stores/FeedbackStore.spec.js]: Done!

**Pull request merged**